libs/template/chunk/src/lib/chunk.directive.ts

```diff
@@ -0,0 +1,174 @@
+ * - zone agnostic
+ * - suspense template
+ *
+ * @docsCategory ChunkDirective
+ * @docsPage ChunkDirective
+ * @publicApi
+ */
+@Directive({ selector: '[rxChunk]' })
+export class ChunkDirective implements OnInit, OnDestroy {
+  /**
+   * @description
+   * The rendering strategy with which the chunk directive should schedule the view
+   * creation. If no strategy is defined, `*rxChunk` will use the configured
+   * `primaryStrategy` as fallback instead.
+   *
+   * @example
+   * \@Component({
+   *   selector: 'app-root',
+   *   template: `
+   *     <div *rxChunk="strategy">
+   *       chunked template
+   *     </div>
+   *   `
+   * })
+   * export class AppComponent {
+   *   strategy = 'low';
+   * }
+   *
+   * @param { RxStrategyNames<string> } strategy
+   */
+  @Input('rxChunk') strategy = this.strategyProvider.primaryStrategy;
+
+  /**
+   * @description
+   * Setting the `patchZone` to a falsy value will cause the `*rxChunk` directive
+   * to create the `EmbeddedView` outside of `NgZone`.
+   *
+   * @example
+   * \@Component({
+   *   selector: 'app-root',
+   *   template: `
+   *     <div *rxChunk="patchZone: false">
+   *       chunked template out of NgZone
+   *     </div>
+   *   `
+   * })
+   * export class AppComponent { }
+   *
+   * @param { boolean } patchZone
+   */
+  @Input('rxChunkPatchZone') patchZone = this.strategyProvider.config.patchZone;
+
+  /**
+   * @description
+   * A template to show while the chunk directive is waiting for scheduled task
+   * with the given `RenderStrategy`.
+   * This can be useful in order to minimize layout shifts caused by the delayed
+   * view creation.
+   *
+   * @example
+   * <app-hero [hero]="hero"
+   *  *rxChunk="suspenseTpl: suspenseTemplate"></app-hero>
+   * <ng-template #suspenseTemplate>
+   *   <progress-spinner></progress-spinner>
+   * </ng-template>
+   *
+   * @param {TemplateRef<unknown>} suspense
+   */
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input('rxChunkSuspenseTpl') suspense: TemplateRef<unknown>;
+
+  /** @internal */
+  private _renderObserver: NextObserver<void>;
+
+  /**
+   * @description
+   * A callback informing about when the template was actually created. This can
+   * be used when you need to know the exact timing when the template was
+   * added to the DOM, e.g. for height calculations and such.
+   *
+   * @example
+   * \@Component({
+   *   selector: 'app-root',
+   *   template: `
+   *     <div *rxChunk="renderCallback: renderCallback">
+   *       chunked template
+   *     </div>
+   *   `
+   * })
+   * export class AppComponent {
+   *   renderCallback = new Subject<void>();
+   *
+   *   constructor() {
+   *     this.renderCallback.subscribe(() => {
+   *       // the div is now accessible
+   *     })
+   *   }
+   * }
+   *
+   *
+   * @param {NextObserver<void>} renderCallback
+   */
+  @Input('rxChunkRenderCallback')
+  set renderCallback(renderCallback: NextObserver<void>) {
```

Comment:
Is it possible to use not only subjects but also just a function? I know it's more like a general question for render callbacks but would it worth the try?

Reply:
I like the idea, I would even more like to just have an `@Output`. But it looks like it's not supported, at least for the `*` syntax. https://github.com/angular/angular/issues/12121